src/components/tasklist/Tasklist.js

```diff
@@ -26,50 +25,40 @@ class Tasklist extends Component {
     inProgress = () => {
         console.log('in progress')
     }
-    completeTask = (key, e) => {
-        const userTasks = this.state.tasks.map(
-            (tasks) => {
-                tasks.map((el) => {
-                        if (!el.done) {
-                            el.id === key ? el.done = true : '';
-                        } else if (el.done) {
-                            el.id === key ? el.done = false : '';
-                        }
-                        return el
-                    }
-                )
-                return tasks
-            }
-        );
-        this.setState({ tasks: userTasks });
+    completeTask = (outIndex, innerIndex, e) => {
+        const { tasks } = this.state;
+        tasks[outIndex][innerIndex].done === false ?
+            tasks[outIndex][innerIndex].done = true :
+            tasks[outIndex][innerIndex].done = false;
+
+        this.setState({ tasks });
```

Comment:
note: it won't work if you use PureComponent (because array `tasks` is the same)